File: Daily Challanges/2026-04-April/2026-04-24.js
function compress(str) {
  let found = {};
  let compressed = [];
  let words = str.split(" ");
  for (let i = 0; i < words.length; i++) {
    const word = words[i];
    if (word in found) {
      compressed.push(found[word]);
    } else {
      found[word] = String(i + 1);
      compressed.push(word);
    }
  }
  return compressed.join(" ");
}

console.log(
  compress("practice makes perfect and perfect practice makes perfect"),
);
console.log(compress("hello hello hello"));

File: Daily Challanges/2026-04-April/2026-04-24.py
def compress(s):
    found = {}
    compressed = []
    for i, word in enumerate(s.split(' ')):
        if word in found:
            compressed.append(found[word])
        else:
            found[word] = str(i+1)
            compressed.append(word)
    return ' '.join(compressed)


print(compress("practice makes perfect and perfect practice makes perfect"))
print(compress("hello hello hello"))